dom.bs

```diff
@@ -6667,18 +6683,32 @@ an optional null or string <var>prefix</var> (default null), and an optional nul
 <var>namespace</var> (default null):
 
 <ol>
- <li>Let <var>attribute</var> be the result of
+ <li><p>Let <var>attribute</var> be the result of
  <a lt="get an attribute by namespace and local name">getting an attribute</a> given
  <var>namespace</var>, <var>localName</var>, and <var>element</var>.
 
- <li>If <var>attribute</var> is null, create an <a>attribute</a> whose <a for=Attr>namespace</a> is
- <var>namespace</var>, <a for=Attr>namespace prefix</a> is <var>prefix</var>,
- <a for=Attr>local name</a> is <var>localName</var>, <a for=Attr>value</a> is <var>value</var>, and
- <a for=Node>node document</a> is <var>element</var>'s <a for=Node>node document</a>, then
- <a lt="append an attribute">append</a> this <a>attribute</a> to <var>element</var>, and then
- return.
+ <li><p>If <var>attribute</var> is null, then set attribute to an <a>attribute</a> whose
+ <a for=Attr>namespace</a> is <var>namespace</var>, <a for=Attr>namespace prefix</a> is
+ <var>prefix</var>, <a for=Attr>local name</a> is <var>localName</var>, <a for=Attr>value</a> is
+ <var>value</var>, and <a for=Node>node document</a> is <var>element</var>'s
+ <a for=Node>node document</a>.
+
+ <li><p>Let <var>verifiedValue</var> be the result of
+ <a lt="verify attribute value">verifying an attribute value</a> given <var>value</var>,
+ <var>attribute</var>, and <var>element</var>.
```

Comment:
Hmm, why do we need this algorithm to verify the value? I'd expect the relevant webidl attribute setters to be modified, similarly to set HTMLScriptElement.src.

In fact, HTMLIFrameElement.srcdoc setter already does TT check to get a reasonable string and this would do it again? Or am I missing something.

Reply:
But this is for setAttributeNS()? Does this also get used by property setters?

Reply:
Ah yeah you're right this is shared by element reflection...

So yeah this will need moving out to the method algorithm. I think in this case it's probably different enough to just duplicate it a bit.